File: 3-01-aiAidedCoding/src/main/java/StringAndPrimeUtilities.java
import java.util.ArrayList;
import java.util.List;

public class StringAndPrimeUtilities {

    // Method to reverse a given string
    public static String reverseString(String input) {
        StringBuilder reversedString = new StringBuilder(input);
        return reversedString.reverse().toString();
    }

    // Method to check if a string is a palindrome
    public static boolean isPalindrome(String input) {
        String reversedString = reverseString(input);
        return input.equals(reversedString);
    }

    // Method to generate the first n prime numbers
    public static List<Integer> generatePrimes(int n) {
        List<Integer> primes = new ArrayList<>();
        int num = 2; // Start with the first prime number
        while (primes.size() < n) {
            if (isPrime(num)) {
                primes.add(num);
            }
            num++;
        }
        return primes;
    }

    // Helper method to check if a number is prime
    private static boolean isPrime(int num) {
        if (num <= 1) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: 3-01-aiAidedCoding-kotlin/src/StringAndPrimeUtilities.kt
class StringAndPrimeUtilities {

    fun reverseString(input: String): String {
        return input.reversed()
    }

    fun isPalindrome(input: String): Boolean {
        return input == reverseString(input)
    }

    fun generatePrimes(n: Int): List<Int> {
        val primes = mutableListOf<Int>()
        var num = 2 // Start with the first prime number
        while (primes.size < n) {
            if (isPrime(num)) {
                primes.add(num)
            }
            num++
        }
        return primes
    }

    fun isPrime(num: Int): Boolean {
        if (num <= 1) {
            return false
        }
        for (i in 2..Math.sqrt(num.toDouble()).toInt()) {
            if (num % i == 0) {
                return false
            }
        }
        return true
    }
}

// Main function to test the methods
fun main() {
    val utilities = StringAndPrimeUtilities()

    // Test reverseString method
    val testString = "Hello"
    println("Reversed String: ${utilities.reverseString(testString)}")

    // Test isPalindrome method
    val palindromeString = "anna"
    println("Is Palindrome: ${utilities.isPalindrome(palindromeString)}")

    // Test generatePrimes method
    val n = 10
    println("First $n prime numbers: ${utilities.generatePrimes(n)}")
}
